Source/KerbalConstructionTime/Utilities/Utilities.cs

```diff
@@ -2122,6 +2126,131 @@ public static void CleanupDebris(string launchSiteName)
                 }
             }
         }
+
+        public static Tuple<float, List<string>> GetVesselUnlockInfo(ShipConstruct ship)
+        {
+            if (HighLogic.CurrentGame.Mode != Game.Modes.CAREER) return new Tuple<float, List<string>>(0, new List<string>());
+
+            // filter the ship parts list to those parts that are not already purchased
+            IEnumerable<KeyValuePair<AvailablePart, PartPurchasability>> purchasableParts = GetPartsWithPurchasability(ship.Parts).Where(kvp => kvp.Value.Status == PurchasabilityStatus.Purchasable || kvp.Value.Status == PurchasabilityStatus.Unavailable);
+            List<string> ecmPartsList = new List<string>();
+            float runningCost = 0;
+
+            // compare the part specified entry cost to the ECM database
+            foreach (AvailablePart part in purchasableParts.Select(kvp => kvp.Key))
+            {
+                int rawCost = part.entryCost;
+                double ecmEstCost = RealFuels.EntryCostManager.Instance.ConfigEntryCost(part.name);
+                if (rawCost == ecmEstCost)
+                {
+                    // this part is managed by the ECM, save its name later for a batch request
+                    string sanitizedName = RealFuels.Utilities.SanitizeName(part.name);
```

Comment:
In looking through the RF ECM code, turns out it sanitizes for you so locally sanitizing is unnecessary.

Reply:
I think we saw that of the various RF calls I make, not every one of them sanitizes. And I have to make sure I have a matchable name in my local collections, so I sanitize on this side.